File: app/src/main/java/de/karzek/diettracker/presentation/main/cookbook/recipeManipulation/adapter/viewHolder/RecipeManipulationItemAddViewHolder.java
package de.karzek.diettracker.presentation.main.cookbook.recipeManipulation.adapter.viewHolder;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.ImageView;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import de.karzek.diettracker.R;
import de.karzek.diettracker.presentation.main.cookbook.recipeManipulation.adapter.itemWrapper.RecipeManipulationViewItemWrapper;

/**
 * Created by MarjanaKarzek on 30.05.2018.
 *
 * @author Marjana Karzek
 * @version 1.0
 * @date 30.05.2018
 */
public class RecipeManipulationItemAddViewHolder extends RecyclerView.ViewHolder {

    private final OnAddManualIngredientClickListener onAddManualIngredientClickListener;
    private final OnStartGrocerySearchClickListener onStartGrocerySearchClickListener;
    private final OnStartBarcodeScanClickListener onStartBarcodeScanClickListener;

    public RecipeManipulationItemAddViewHolder(ViewGroup viewGroup,
                                               OnAddManualIngredientClickListener onAddManualIngredientClickListener,
                                               OnStartGrocerySearchClickListener onStartGrocerySearchClickListener,
                                               OnStartBarcodeScanClickListener onStartBarcodeScanClickListener) {
        super(LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.viewholder_recipe_man_ingredient_add, viewGroup, false));
        ButterKnife.bind(this, itemView);

        this.onAddManualIngredientClickListener = onAddManualIngredientClickListener;
        this.onStartGrocerySearchClickListener = onStartGrocerySearchClickListener;
        this.onStartBarcodeScanClickListener = onStartBarcodeScanClickListener;
    }

    public void bind(RecipeManipulationViewItemWrapper item) {

    }

    @OnClick(R.id.add_ingredient) public void onAddManualIngredientClicked() {
        onAddManualIngredientClickListener.onAddManualIngredientClicked();
    }

    @OnClick(R.id.action_search) public void onStartGrocerySearchClicked() {
        onStartGrocerySearchClickListener.onStartGrocerySearchClicked();
    }

    @OnClick(R.id.action_barcode_search) public void onStartBarcodeScanClicked() {
        onStartBarcodeScanClickListener.onStartBarcodeScanClicked();
    }

    public interface OnAddManualIngredientClickListener {
        void onAddManualIngredientClicked();
    }

    public interface OnStartGrocerySearchClickListener {
        void onStartGrocerySearchClicked();
    }

    public interface OnStartBarcodeScanClickListener {
        void onStartBarcodeScanClicked();
    }

}

File: app/src/main/java/de/karzek/diettracker/presentation/main/cookbook/recipeManipulation/adapter/viewHolder/RecipeManipulationItemAddViewHolder.kt
package de.karzek.diettracker.presentation.main.cookbook.recipeManipulation.adapter.viewHolder

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import de.karzek.diettracker.presentation.main.cookbook.recipeManipulation.adapter.itemWrapper.RecipeManipulationViewItemWrapper
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.viewholder_recipe_man_ingredient_add.*

/**
 * Created by MarjanaKarzek on 30.05.2018.
 *
 * @author Marjana Karzek
 * @version 1.0
 * @date 30.05.2018
 */
class RecipeManipulationItemAddViewHolder(override val containerView: View,
                                          private val onAddManualIngredientClickListener: OnAddManualIngredientClickListener,
                                          private val onStartGrocerySearchClickListener: OnStartGrocerySearchClickListener,
                                          private val onStartBarcodeScanClickListener: OnStartBarcodeScanClickListener) : RecyclerView.ViewHolder(containerView), LayoutContainer {

    fun bind(item: RecipeManipulationViewItemWrapper) {
        setupClickListeners()
    }

    private fun setupClickListeners() {
        add_ingredient.setOnClickListener { onAddManualIngredientClickListener.onAddManualIngredientClicked() }
        action_search.setOnClickListener { onStartGrocerySearchClickListener.onStartGrocerySearchClicked() }
        action_barcode_search.setOnClickListener { onStartBarcodeScanClickListener.onStartBarcodeScanClicked() }
    }

    interface OnAddManualIngredientClickListener {
        fun onAddManualIngredientClicked()
    }

    interface OnStartGrocerySearchClickListener {
        fun onStartGrocerySearchClicked()
    }

    interface OnStartBarcodeScanClickListener {
        fun onStartBarcodeScanClicked()
    }

}
